packages/query-core/src/infiniteQueryBehavior.ts

```diff
@@ -134,6 +134,11 @@ export function infiniteQueryBehavior<
           // Fetch remaining pages
           for (let i = 1; i < oldPages.length; i++) {
             promise = promise.then((pages) => {
+              context.dispatch({
```

Comment:
`i` is the value for previous page also in every promise I changed fetchPages for previous one